### PR TITLE
Check constraints on commit

### PR DESCRIPTION
**Pull request by Hydrocharged**

### CODE REVIEW

File: bats/foreign-keys.bats [hunk 1]
```diff
@@ -962,7 +977,7 @@ SET FOREIGN_KEY_CHECKS=0;
 UPDATE parent SET v1 = v1 - 1;
 SQL
     dolt add -A
-    dolt commit -m "updated parent"
+    dolt commit --force -m "updated parent"
```
Comment (timsehn) [inline bats/foreign-keys.bats:980]: Why do we need these --forces now?

Comment (Hydrocharged) [reply bats/foreign-keys.bats:980]: Previously we didn't check row data on commit (only schema information), but now we do. These tests are verifying that the merge behavior is working correctly when working with invalid table state, so now to commit bad tables we have to force them.

Comment (timsehn) [reply bats/foreign-keys.bats:980]: Understood

Comment (timsehn) [reply bats/foreign-keys.bats:980]: I see. All these have `SET FOREIGN_KEY_CHECKS=0;` above.

**Review by timsehn — APPROVED** [review]

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM.

This is a good safety measure for now, but we need come up with a better, more efficient way to validate FKs. Doing a full scan over two indexes for every Foreign Key is prohibitively expensive. We should think about how to leverage diffs to make this more performant.

**Comment by Hydrocharged** [comment]
> 
> 
> LGTM.
> 
> This is a good safety measure for now, but we need come up with a better, more efficient way to validate FKs. Doing a full scan over two indexes for every Foreign Key is prohibitively expensive. We should think about how to leverage diffs to make this more performant.

I'll add a TODO and make the suggestion so we know to come back to it later.

**Pull request merged**